torch/_dynamo/source.py

```diff
@@ -350,7 +350,14 @@ def guard_source(self):
         return self.base.guard_source()
 
     def name(self):
-        return f"___odict_getitem({self.base.name()}, {self.index!r})"
+        if isinstance(self.index, type):
+            if self.index.__module__ == 'builtins':
+                rep = self.index.__qualname__
+            else:
+                rep = f"{self.index.__module__}.{self.index.__qualname__}"
```

Comment:
Understood. I know that I made numpy work by adding it to `CLOSURE_VARS`, but I'm still not sure how to make variables carry arbitray exotic modules with them and load them during execution.

Reply:
I would add a module loader function to the scope, and then call `importlib.import_module("numpy")`, with carveouts for important modules (like numpy) so you can avoid the function call otherwise.